### DISCUSSION TITLE
limit returned fields with auth-with-password

### DISCUSSION THREAD

**Discussion by maxNodeJs** [pocketbase/pocketbase#2983] [Q&A]
Hi, 

Login a user:

const logged_user = await ApiClient.collection("users").authWithPassword(email, password, {}, {
                fields: "id,email,alias_name,tenant,avatar,created,updated",
            })

The above returns empty object, and when I remove the fields params it returns the full user object with token.
            
I have also tried below example using insomnia and it also returns empty object.
Removing the fields params returns the full user object as expected. same as above.

POST request to end point: http://127.0.0.1:8090/api/collections/users/auth-with-password?fields=id,created

I am using pocketbase 0.16.6

Just as a note, when sending get request to any collection with fields params returns the correct fields, it is only on authwithpassword I get empty object.

What am I doing wrong here?

**Comment by ganigeorgiev** [comment]
This is not documented very well at the moment as the `fields` behavior may change in the future, but the `fields` format for any non-list/search response should correspond to the returned JSON, aka. in you case this should be something like: `fields=token,record.id,record.created`, but keep in mind that the above response will affect what data will be stored in `pb.authStore`.

**Reply by maxNodeJs** [reply]
Great thank you for your quick response. I am aware of the pb.authStore.

**Comment by laszlo1337** [comment]
Hello. I need help with this. I am authenticating using this method:
```
const response = await pb.collection('users').authWithPassword(email, password, { fields: 'token,record.email,record.id,record.role,record.verified' })
```

And in this response I do have the fields I actually chosen here, but in the actual cookie there are still all fields present.
Even if I manually save the model to authStore using `pb.authStore.save(response.token, response.record)`

To `loadFromCookie` and `exportToCookie` I provide custom cookie key, but this seems to work.

EDIT: I use PB v0.19.3

**Comment by laszlo1337** [comment]
I was looking some more but haven't found any mentions of this. Maybe I need custom store like here? https://github.com/pocketbase/js-sdk#custom-auth-store tho I don't know what would it change since as I understand upon successful auth the authStore is automatically saving the returned model.

**Reply by ganigeorgiev** [reply]
You don't need custom store, nor manually calling `save`.

Without a code sample showing what you are doing it is difficult to guess what is the problem in your case.

Double check any `authRefresh`, `loadFromCookie`, etc. calls.

**Reply by laszlo1337** [reply]
Ok then now I am fairly sure this would be the fault of authRefresh. I wasn't aware this takes `fields:` option as well.
